sounds/models.py

```diff
@@ -478,106 +468,86 @@ def bulk_query_solr(self, sound_ids):
-          sound.num_ratings,
-          sound.description,
-          sound.moderation_state,
-          sound.processing_state,
-          sound.processing_ongoing_state,
-          sound.similarity_state,
-          sound.created,
-          sound.num_downloads,
-          sound.num_comments,
-          sound.pack_id,
-          sound.duration,
-          sounds_pack.name as pack_name,
-          sound.license_id,
-          sounds_license.name as license_name,
-          sounds_license.deed_url as license_deed_url,
-          sound.geotag_id,
-          geotags_geotag.lat as geotag_lat,
-          geotags_geotag.lon as geotag_lon,
-          geotags_geotag.location_name as geotag_name,
-          tickets_ticket.key as ticket_key,
-          sounds_remixgroup_sounds.id as remixgroup_id,
-          accounts_profile.has_avatar as user_has_avatar,
-          %s
-          %s
-          %s
-          ARRAY(
-            SELECT tags_tag.name
-            FROM tags_tag
-            LEFT JOIN tags_soundtag ON tags_soundtag.sound_id = sound.id
-          WHERE tags_tag.id = tags_soundtag.tag_id) AS tag_array
-        FROM
-          sounds_sound sound
-          LEFT JOIN auth_user ON auth_user.id = sound.user_id
-          LEFT JOIN accounts_profile ON accounts_profile.user_id = sound.user_id
-          LEFT JOIN sounds_pack ON sound.pack_id = sounds_pack.id
-          LEFT JOIN sounds_license ON sound.license_id = sounds_license.id
-          LEFT JOIN geotags_geotag ON sound.geotag_id = geotags_geotag.id
-          LEFT JOIN tickets_ticket ON tickets_ticket.sound_id = sound.id
-          %s
-          LEFT OUTER JOIN sounds_remixgroup_sounds ON sounds_remixgroup_sounds.sound_id = sound.id
-        WHERE %s """ % (self.get_search_engine_similarity_state_sql(),
-                        self.get_analysis_state_essentia_exists_sql(),
-                        self.get_analyzers_data_select_sql() if include_analyzers_output else '',
-                        self.get_analyzers_data_left_join_sql() if include_analyzers_output else '',
-                        where, )
-        if order_by:
-            query = f"{query} ORDER BY {order_by}"
-        if limit:
-            query = f"{query} LIMIT {limit}"
-        return self.raw(query, args)
 
+    def bulk_query(self, include_analyzers_output=False):
+        tags_subquery = Tag.objects.filter(soundtag__sound=OuterRef("id")).values("name")
+        analysis_subquery = SoundAnalysis.objects.filter(
+            sound=OuterRef("id"), analyzer=settings.FREESOUND_ESSENTIA_EXTRACTOR_NAME, analysis_status="OK"
+        )
+        search_engine_similarity_subquery = SoundAnalysis.objects.filter(
+            sound=OuterRef("id"), analyzer=settings.SEARCH_ENGINE_DEFAULT_SIMILARITY_ANALYZER, analysis_status="OK"
+        )
+
+        qs = self.select_related(
+            'user', 'user__profile', 'license', 'ticket', 'pack', 'geotag'
+            ).annotate(
+                username=F("user__username"),
```

Comment:
these annotate values are the values that the old bulk query returned. This makes it easy to run the code as-is without changing it to reference the related objects. We can slowly move them out

Reply:
removed most of them!